### PR TITLE
feat(chain-learning): prompt-chain learning system with 5 ONEX nodes [OMN-7032]

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary

- Adds a chain-learning cache layer: verified workflow trajectories are stored in Qdrant and replayed on similar future prompts, shortcutting full re-exploration
- Implements 5 ONEX nodes following the four-node pattern: retrieval effect (embed + Qdrant query), replay compute (context adaptation), verify reducer (FSM lifecycle), store effect (Qdrant upsert), and orchestrator (workflow coordination)
- Includes 87 unit tests covering all models, handlers, contract YAML validation, FSM transitions, error handling, and edge cases

## Architecture

```
node_chain_orchestrator (ORCHESTRATOR)
  |
  +--> node_chain_retrieval_effect (EFFECT) -- embed prompt, query Qdrant
  |
  +--> node_chain_replay_compute (COMPUTE) -- adapt cached chain to new context
  |
  +--> node_chain_verify_reducer (REDUCER) -- FSM: pending->retrieving->[replaying|exploring]->verifying->[complete|fallback]
  |
  +--> node_chain_store_effect (EFFECT) -- store verified chain to Qdrant
```

## Test plan

- [x] All 87 unit tests pass (`uv run pytest tests/unit/nodes/node_chain_learning/ -v`)
- [x] Pre-commit hooks pass (all architecture, contract, pattern, naming validators)
- [x] mypy strict passes on all new code
- [x] Models: frozen, extra=forbid, field constraints, round-trip serialization
- [x] Effect handlers: mocked embedding + Qdrant clients, error paths (embedding failure, search failure, invalid payloads)
- [x] Compute handler: exact replay confidence, context substitution, confidence decay
- [x] Reducer: all valid FSM transitions, error from any non-terminal state, invalid transition rejection, terminal state rejection
- [x] Contracts: required fields, correct node_type, handler routing, event bus topics, FSM definition, workflow coordination

## Summary by CodeRabbit

* **New Features**
  * Added a complete chain-learning workflow: retrieval, replay/adaptation, verification, orchestration, and persistent storage with event-driven contracts and handlers.
* **Chores**
  * Increased infrastructure union-validation limit.
* **Tests**
  * Added comprehensive unit tests for contracts, handlers, compute/reducer logic, and models covering validation, workflows, and failure modes.

### CODE REVIEW

File: src/omnibase_infra/nodes/node_chain_orchestrator/handlers/handler_chain_replay_complete.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                "Replay accepted: confidence=%.2f, steps=%d (correlation_id=%s)",
+                replay_result.confidence,
+                len(replay_result.adapted_steps),
+                correlation_id,
+            )
+            return {
+                "action": "store",
+                "confidence": str(replay_result.confidence),
+                "summary": replay_result.adaptation_summary,
+            }
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_infra/nodes/node_chain_orchestrator/handlers/handler_chain_replay_complete.py]: Deferred: The verify-before-store routing is a design decision for this chain-learning system. The confidence threshold acts as the implicit verification gate for replay paths. A dedicated verify phase can be added as a follow-up once the system's validation requirements are validated end-to-end.

File: src/omnibase_infra/nodes/node_chain_orchestrator/models/model_chain_replay_input.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from .model_chain_entry import ModelChainEntry
+
+
+class ModelChainReplayInput(BaseModel):
+    """Input to the chain replay compute node."""
+
+    model_config = ConfigDict(frozen=True, extra="forbid")
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_infra/nodes/node_chain_orchestrator/models/model_chain_replay_input.py]: Deferred: `from_attributes=True` missing from ConfigDict. Will add in a follow-up alongside the other chain models to achieve consistent guideline compliance.

File: src/omnibase_infra/nodes/node_chain_orchestrator/models/model_chain_store_request.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from .model_chain_entry import ModelChainEntry
+
+
+class ModelChainStoreRequest(BaseModel):
+    """Request to store a verified chain in Qdrant."""
+
+    model_config = ConfigDict(frozen=True, extra="forbid")
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_infra/nodes/node_chain_orchestrator/models/model_chain_store_request.py]: Deferred: `from_attributes=True` missing from ConfigDict. Will add in a follow-up alongside the other chain models to achieve consistent guideline compliance.
